### DISCUSSION TITLE
Difference between bbox and mask filtering

### DISCUSSION THREAD

**Discussion by Reslan-Tinawi** [geopandas/geopandas#2927] [Q&A] [answered]
Hello GeoPandas community 👋

I was exploring the [Ookla speedtest data](https://github.com/teamookla/ookla-open-data) and wanted to create a download speed choropleth map for my country, Syria.

The data is not geographically partitioned (one file for the entire world), and it's relatively big (about 3M rows). Reading the entire file takes about 30 minutes. It consist of small rectangular tiles (approximately 610.8 meters by 610.8 meters)

I wanted to leverage the filtering techniques in `read_file` (the `bbox` and `mask` arguments), to **only** select tiles that intersects with a region of interest (my country's polygon).

This worked fine, and delivered good time improvement:

- 16 seconds using `bbox` filtering
- 2 minutes using `mask` filtering

However, I realized that filtering with `bbox` returns more results than filtering with `mask` (and takes much less time).

I created this [(reproducible) notebook](https://colab.research.google.com/drive/1upuofCtZ-HucVrgoPBPaHZ_kjkiqPyqB?usp=sharing) on Colab to understand how these two approaches work. I came to the conclusion that when using `bbox`, the `envelope` of the geometry is actually being used for the filtering, and therefore, it returns some tiles that don't intersect with the specified region of interest. In contrast, using `mask` is more strict and returns **only** the tiles that actually intersect with the specified region of interest.

Is my understanding correct? If so, I believe that the documentation of the `read_file` function could be improved (I'd be happy to contribute), and also, adding example to this use case (**Working with large datasets**, something similar to [Pandas: Scaling to large datasets](https://pandas.pydata.org/docs/user_guide/scale.html) ) would be very useful for future reference (I spent the last two days figuring out how to achieve this 😅)

**Comment by martinfleis** [comment, marked as answer]
Hi, yes, your observation is correct. When passing a geometry to the `bbox` keyword, the bounding box of the geometry is used. Originally, you had to pass the actual tuple of `(minx, miny, maxx, maxy)` coordinates. today, we support passing geometry or a geoseries as a convenient way that does not require manual extraction of the bounding box.

What is actually happening is nicely described [here in GDAL docs](https://gdal.org/api/vector_c_api.html#_CPPv426OGR_L_SetSpatialFilterRect9OGRLayerHdddd), which is used under the hood by both Fiona and pyogrio.

> Only features that geometrically intersect the given rectangle will be returned.

I suppose we can make this behaviour clearer in the docs. If you want to make a pull request, it would be most welcome. Same applies to the example :).
